06.bowling_object/frame.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+
+class Frame
+  LAST_FRAME = 9
+  MAX_SHOTS_IN_LAST_FRAME = 3
+  MAX_SHOTS_IN_REGULAR_FRAME = 2
+
+  attr_reader :scores
+
+  def initialize(frame_idx)
+    @scores = []
+    @frame_idx = frame_idx
+  end
+
+  def add_shot(marks)
+    marks.each do |mark|
+      @scores << Shot.new(mark)
+      break if complete?
+    end
+    @scores.size
+  end
+
+  def score(frames)
+    @frame_idx != LAST_FRAME ? @scores.sum(&:score) + bonus_score(frames) : @scores.sum(&:score)
+  end
+
+  private
+
+  def complete?
+    if @frame_idx != LAST_FRAME
+      strike? || @scores.size == MAX_SHOTS_IN_REGULAR_FRAME
+    else
+      @scores.size == (strike? || spare? ? MAX_SHOTS_IN_LAST_FRAME : MAX_SHOTS_IN_REGULAR_FRAME)
```

Comment:
ここは最終フレーム用の処理なのに、`MAX_SHOTS_IN_REGULAR_FRAME`という定数が登場するのはちょっと違和感がありますね。

そもそも`MAX_SHOTS_IN_REGULAR_FRAME`や`MAX_SHOTS_IN_LAST_FRAME`みたいな定数定義はなくして、

```suggestion
      @scores.size == (strike? || spare? ? 3 : 2)
```

みたいにしてもいいんじゃないかなーと思います。（人によって意見はわかれるかもしれませんが）